src/backend/expungeservice/endpoints/users.py

```diff
@@ -2,24 +2,50 @@
 from flask import request, abort, jsonify
 from werkzeug.security import generate_password_hash
 
-# TODO temp hack - replace with table from DB
-users = {}
+from flask import g
+from expungeservice.database.user import create_user, get_user_by_email
+from expungeservice.endpoints.auth import auth_required
+from psycopg2.errors import UniqueViolation
+
 
 class Users(MethodView):
+    @auth_required
     def post(self):
         data = request.get_json()
 
         if data == None:
             abort(400)
 
-        # TODO temp hack - replace with table from DB
-        if data['email'] in users:
-            return 'User already exists! Please login', 202
+        """
+        - the @auth_required decorator identifies the logged in user. If no auth, return 401 UNAUTHORIZED.
+        - Verify the logged in user is admin. if not, return 403 FORBIDDEN.
+        - Verify the password meets the 8-character length requirement. If not, return 422
+        - Attempt to insert the user into the database. If failure due to duplicate entry, return 422
+        - If success, return the new user's email, admin flag, and creation timestamp.
+        """
+
+        authorized_user_db_result = get_user_by_email(g.database, g.auth_hdr_payload['sub'])
+        if not authorized_user_db_result['admin']:
+            return 'Logged in user not admin', 403
+
+        if len(data['password']) <8:
```

Comment:
This will throw a KeyError if there is no `'password'` entry in `data`, right? Just seems worthwhile to comment, since this would be a client error.

We'd ideally want a general method to use in Controllers that, given a dict and a list of required keys, checks the dict to ensure those keys are present.

Reply:
@maxkwallace makes a good point here. Any time a `POST` or `PUT` body comes in we should check it for the required fields.

Reply:
@maxkwallace I added a function  to request.py to use for checking required data field in the request. Let me know if you think there's a better place to put this.

Reply:
I don't know the conventions / inheritance structure well enough for this code to comment authoritatively on where to put it. I think having it in `request.py` is probably fine since it's easy to move later if we need to.